packages/environment/test/develop.js

```diff
@@ -29,14 +29,10 @@ describe("Environment develop", function () {
         gas: expectedNetwork.gas
       }
     };
-    sinon.stub(Environment, "detect");
-  });
-
-  afterEach("Restore Environment detect", async function () {
-    Environment.detect.restore();
   });
 
-  it("Environment.develop overwrites the network_id of the network", async function() {
+  it("Environment.develop overwrites the network_id of the network", async function () {
+    sinon.stub(Environment, "detect");
```

Comment:
I'm confused why this is being stubbed... what is this for?

Reply:
`Environment.develop()` invokes `Environment.detect()` as its final step, which isn't needed to validate the test's assertion.  I'll add a clarifying comment and leave potential refactoring for followup work.

Reply:
Oh, huh, OK.  Odd use and I'd suggest getting rid of it, but I guess it's OK to put that off? <shrug>

Reply:
This is one of those things that touches TruffleConfig's side-effect behavior. There are a bunch of methods in `Environment` that modify `TruffleConfig` :/ . A code smell to be addressed later.